pkg/m3db/reconcile.go

```diff
@@ -0,0 +1,21 @@
+package m3db
+
+import (
+	api "github.com/m3db/m3db-operator/pkg/apis/m3db/v1alpha1"
+)
+
+// Reconcile reconciles the vault cluster's state to the spec specified by vr
+// by preparing the TLS secrets, deploying the etcd and vault cluster,
+// and finally updating the vault deployment if needed.
+func Reconcile(s *api.M3DBService) (err error) {
```

Comment:
Why does this function have a named return value for the error, but others do not?

Reply:
Updated, thanks for catching the inconsistency. IMHO I favor named return values but going to omit it for the time being.